File: javascript/functions/json_operations.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

const data = {
    name: 'John Doe',
    age: 30,
    skills: ['JavaScript', 'Node.js', 'React'],
    active: true
};

const jsonString = JSON.stringify(data, null, 2);
console.log(`JSON String:\n${jsonString}`);

const parsedData = JSON.parse(jsonString);
console.log(`\nParsed data: ${JSON.stringify(parsedData)}`);
console.log(`Name: ${parsedData.name}`);
console.log(`Skills: ${parsedData.skills.join(', ')}`);

const jsonFile = path.join(__dirname, '../tmp/data.json');
fs.writeFileSync(jsonFile, jsonString);
console.log(`\nWritten to ${jsonFile}`);

const loadedData = JSON.parse(fs.readFileSync(jsonFile, 'utf8'));
console.log(`Loaded from file: ${JSON.stringify(loadedData)}`);

File: ruby/functions/json_operations.rb
#!/usr/bin/env ruby

require 'json'

data = {
  name: 'John Doe',
  age: 30,
  skills: ['Ruby', 'Rails', 'JavaScript'],
  active: true
}

json_string = JSON.pretty_generate(data)
puts "JSON String:\n#{json_string}"

parsed_data = JSON.parse(json_string)
puts "\nParsed data: #{parsed_data}"
puts "Name: #{parsed_data['name']}"
puts "Skills: #{parsed_data['skills'].join(', ')}"

json_file = '../tmp/data.json'
File.write(json_file, json_string)
puts "\nWritten to #{json_file}"

loaded_data = JSON.parse(File.read(json_file))
puts "Loaded from file: #{loaded_data}"
